fuzzing/private/regression.bzl

```diff
@@ -61,6 +68,16 @@ Executes a fuzz test on its seed corpus.
             cfg = "target",
             mandatory = True,
         ),
+        "_lcov_merger": attr.label(
+            default = "@bazel_tools//tools/test:lcov_merger",
+            executable = True,
+            cfg = "host",
+        ),
+        "_collect_cc_coverage": attr.label(
+            default = "@bazel_tools//tools/test:collect_cc_coverage",
+            executable = True,
+            cfg = "host",
+        ),
```

Comment:
I think I need to understand this better: Why do we need to add these dependencies? Are these going to be built even in non-coverage mode?

(It would help if you could point me to some documentation discussing this.)

Reply:
[collect_cc_coverage](https://github.com/bazelbuild/bazel/blob/09c621e4cf5b968f4c6cdf905ab142d5961f9ddc/tools/test/collect_cc_coverage.sh) runs the gcov/llvm-cov commands to gather coverage.

[lcov_merger](https://github.com/bazelbuild/bazel/tree/09c621e4cf5b968f4c6cdf905ab142d5961f9ddc/tools/test/CoverageOutputGenerator/java/com/google/devtools/coverageoutputgenerator) merges different coverage output files into a single output file for consumption.

I found these dependencies from the [implementation of the native cc_test rule](https://github.com/bazelbuild/bazel/blob/09c621e4cf5b968f4c6cdf905ab142d5961f9ddc/src/main/java/com/google/devtools/build/lib/bazel/rules/cpp/BazelCcTestRule.java#L47).

They will be built in non-coverage mode. Native rules with coverage support use late-bound attributes to avoid building these dependencies unnecessarily, but that's [not yet an option in Starlark](https://github.com/bazelbuild/bazel/issues/8736).

Reply:
Unfortunately I'm reluctant to accept this change as-is. I understand this is a Starlark limitation, but the workaround of including these dependencies on all builds is inadequate IMO.

Let me do some additional research to better understand all of our alternatives. If coverage support is not mature enough yet in Bazel, I would postpone the addition of this feature for now.

Reply:
This should be fixed by https://github.com/bazelbuild/bazel/pull/13983. Until then, I don't know of a better solution.

Reply:
[The underlying issue](https://github.com/bazelbuild/bazel/issues/13978) is now a release blocker for Bazel 5, so it's likely that `lcov_merger` will be provided by default in a way that it would be built only with `bazel coverage`. The same does not apply to `collect_cc_coverage`, but that is merely a shell script and thus causes no compilation overhead.

Reply:
My fix [will not be accepted](https://github.com/bazelbuild/bazel/pull/13983#issuecomment-981542636), so this will very likely not be possible with Bazel 5.

Reply:
@zhenyudg I haven't forgotten about this. If you want to know whether there has been progress in getting the underlying issue resolved in Bazel, you can track https://github.com/bazelbuild/bazel/issues/8736.

Reply:
@zhenyudg While working on the upstream fix, I may have found a workaround for Bazel 4+: Use an `alias` as the default for `_lcov_merger` and in that alias, use a `select` on a [`config_setting`](https://docs.bazel.build/versions/main/configurable-attributes.html) for [`collect_code_coverage`](https://docs.bazel.build/versions/main/command-line-reference.html#flag--collect_code_coverage) to switch between the real lcov_merger target and a dummy target (e.g., an empty executable file).